ClipboardAPI/clipboard-change-event-explainer.md

```diff
@@ -76,92 +71,102 @@ Web based editors like Word Online, Excel Online, Google Sheets, etc. may suppor
 
-## 4. Example javascript code for detecting clipboard changes:
+## 4. Proposed Approach
+
+### 4.1 Example javascript code for detecting clipboard changes:
 
 ```javascript
-  function callback(event) {
-      // Read clipboard contents using the navigator.clipboard API
-      navigator.clipboard.readText().then(text => console.log(text));
+  // Event handler for clipboardchange event which contains the data types present in clipboard
+  function onClipboardChanged(event) {
+    document.getElementById("text_paste_button").disabled = !(event.types.includes('text/plain'));
+    document.getElementById("html_paste_button").disabled = !(event.types.includes('text/html'));
+    document.getElementById("png_paste_button").disabled = !(event.types.includes('img/png'));
   }
 
-  // Try to read the clipboard to trigger a permissions prompt if required.
-  navigator.clipboard.readText().then(() => {
-
-    // Start listening to the clipboardchange event
-    navigator.clipboard.addEventListener("clipboardchange", callback);
-  });
+  // This will trigger a permission popup for "clipboard-types-read" / "clipboard-read" (if choice not already provided)
+  navigator.clipboard.addEventListener("clipboardchange", onClipboardChanged);
 ```
 
-A sample web application which demonstrates the usage of "clipboardchange" event for showing available paste formats for rich web editors [Scenario 2.2](#22-scenario-show-available-paste-formats-in-web-based-editors) can be found [here](./clipboard-change-event-example-app.html).
-
-## 5. Event spec details and open questions
+A sample web application which demonstrates the usage of "clipboardchange" event for showing available paste formats for rich web editors [Scenario 2.2](#21-scenario-show-available-paste-formats-in-web-based-editors) can be found [here](./clipboard-change-event-example-app.html).
 
-### 5.1 User permission requirement
+### 4.2 Permissions and Interop - No user permission required 
+ 
+Today browser engines have different approaches to clipboard API permissions. While Chromium has [this permissions model](https://github.com/w3c/clipboard-apis/blob/main/explainer.adoc#clipboard-permissions) for clipboard, [Firefox](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) and [Safari](https://webkit.org/blog/10855/async-clipboard-api/) rely on combination of user activation and user gesture for web pages to access user clipboard contents. This strict requirement is present because a user's clipboard contents are highly sensitive and can contain private data like passwords, security tokens. However, we will see that core scenarios involving clipboardchange event does not require complete user clipboard contents and hence we propose a more lenient permissions approach, that works in all browsers, for this new API. 
+ 
+**Permission prompt in Chromium**
```

Comment:
Can this analysis be moved to an "appendix" like section at the end of the explainer. It's good information, but not the most important for a reader trying to understand your proposal.